dataactcore/interfaces/function_bag.py

```diff
@@ -118,3 +119,125 @@ def checkNumberOfErrorsByJobId(jobId, errorType='fatal'):
+                Job.file_type_id == fileTypeId,
+                Job.job_type_id == JOB_TYPE_DICT['csv_record_validation']).\
+            one()
+        valJob.job_status_id = JOB_STATUS_DICT['waiting']
+        valJob.original_filename = filename
+        valJob.filename = filePath
+        # Reset file size and number of rows to be set during validation of new file
+        valJob.file_size = None
+        valJob.number_of_rows = None
+        # Delete error metdata this might exist from a previous run of this validation job
+        sess.query(ErrorMetadata).\
+            filter(ErrorMetadata.job_id == valJob.job_id).\
+            delete(synchronize_session='fetch')
+        # Delete file error information that might exist from a previous run of this validation job
+        sess.query(File).filter(File.job_id == valJob.job_id).delete(synchronize_session='fetch')
+
+    else:
+        # create a new record validation job and add dependencies if necessary
+        if fileType == "awardee_attributes":
+            d1ValJob = sess.query(Job).\
+                filter(Job.submission_id == submissionId,
+                       Job.file_type_id == FILE_TYPE_DICT['award_procurement'],
+                       Job.job_type_id == JOB_TYPE_DICT['csv_record_validation']).\
+                first()
+            if d1ValJob is None:
+                raise Exception("Cannot create E job without a D1 job")
+            # Add dependency on D1 validation job
+            d1Dependency = JobDependency(job_id=uploadJob.job_id, prerequisite_id=d1ValJob.job_id)
+            sess.add(d1Dependency)
+
+        elif fileType == "sub_award":
+            # todo: check for C validation job
+            cValJob = sess.query(Job). \
+                filter(Job.submission_id == submissionId,
+                       Job.file_type_id == FILE_TYPE_DICT['award_financial'],
+                       Job.job_type_id == JOB_TYPE_DICT['csv_record_validation']). \
+                first()
+            if cValJob is None:
+                raise Exception("Cannot create F job without a C job")
+            # Add dependency on C validation job
+            cDependency = JobDependency(job_id=uploadJob.job_id, prerequisite_id=cValJob.job_id)
+            sess.add(cDependency)
+
+        else:
+            # E and F don't get validation jobs
+            valJob = Job(original_filename=filename, filename=filePath, file_type_id=fileTypeId,
+                         job_status_id=JOB_STATUS_DICT['waiting'],
+                         job_type_id=JOB_TYPE_DICT['csv_record_validation'],
+                         submission_id=submissionId)
+            sess.add(valJob)
+            sess.flush()
+            # Add dependency between file upload and db upload
+            uploadDependency = JobDependency(job_id=valJob.job_id, prerequisite_id=uploadJob.job_id)
+            sess.add(uploadDependency)
+            jobsRequired.append(valJob.job_id)
+
+        sess.commit()
```

Comment:
Oops, yes...was definitely the intent to commit everything at the end of the function. Thanks for catching that!